pkg/azure/test_utils.go

```diff
@@ -641,6 +697,30 @@ func getFakeNIC() *armnetwork.Interface {
 	}
 }
 
+func getFakeUnattachedNIC() *armnetwork.Interface {
```

Comment:
I don't see where this is used. With getFakeParagliderInterface() and getFakeInterface(), what additional purpose does this serve? But maybe my confusion is just from the basing on your other branches.

Reply:
I mistakenly duplicated a few functions. Fixed it now